# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../slmodem/gcc4.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -Nur slmodem-2.9.10-orig/modem/modem.c slmodem-2.9.10/modem/modem.c
--- slmodem-2.9.10-orig/modem/modem.c	2004-08-08 13:29:19.000000000 +0000
+++ slmodem-2.9.10/modem/modem.c	2006-01-26 18:54:00.854715500 +0000
@@ -1695,10 +1695,10 @@
 	return 0;
 }
 
+static int sregs_init(unsigned char sregs[]);
 
 int modem_reset(struct modem *m)
 {
-	static int sregs_init(unsigned char sregs[]);
 	MODEM_DBG("modem reset...\n");
 	if(m->state != STATE_MODEM_IDLE)
 		modem_hup(m,1);
